### PR TITLE
Add lodepng 20200615

### PR DESCRIPTION
**Pull request by intelligide**
Specify library name and version:  **lodepng/20200615**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/lodepng/all/test_package/test_package.cpp [hunk 1]
```diff
@@ -0,0 +1,123 @@
+#include <iostream>
+#include <vector>
+#include <cstdint>
+#include <cassert>
+
+#include "lodepng.h"
+
+const std::vector<uint8_t> PNG_DATA {
```
Comment (madebr) [inline recipes/lodepng/all/test_package/test_package.cpp:8]: You can put a PNG file in the test folder and open it from arguments.

File: recipes/lodepng/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
```
Comment (madebr) [inline recipes/lodepng/all/conanfile.py:36]: fPIC should also be removed in the `configure` function when `shared` is true.

File: recipes/lodepng/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 2.8.11)
```
Comment (uilianries) [inline recipes/lodepng/all/test_package/CMakeLists.txt:1]:
```suggestion
cmake_minimum_required(VERSION 3.1)
```

File: recipes/lodepng/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
+import os
+import glob
+from conans import ConanFile, CMake, tools
+
+class LodepngConan(ConanFile):
+    name = "lodepng"
+    description = "PNG encoder and decoder in C and C++, without dependencies."
+    license = "ZLIB"
```
Comment (uilianries) [inline recipes/lodepng/all/conanfile.py:8]:
```suggestion
    license = "Zlib"
```

https://spdx.org/licenses/Zlib.html

File: recipes/lodepng/all/test_package/test_package.cpp [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+int main()
+{
+    std::ifstream stream("bees.png", std::ios::in | std::ios::binary);
+    std::vector<uint8_t> data((std::istreambuf_iterator<char>(stream)), std::istreambuf_iterator<char>());
+
+    std::vector<uint8_t> decoded;
+    unsigned width = 0, height = 0;
+    unsigned error = lodepng::decode(decoded, width, height, data);
+    assert(static_cast<bool>(error) == false);
```
Comment (uilianries) [inline recipes/lodepng/all/test_package/test_package.cpp:17]:
```suggestion
```

The idea for test_package is checking the generated package, not the library functionality.

File: recipes/lodepng/all/test_package/test_package.cpp [hunk 6]
```diff
@@ -0,0 +1,22 @@
+#include <iostream>
+#include <vector>
+#include <cstdint>
+#include <cassert>
```
Comment (uilianries) [inline recipes/lodepng/all/test_package/test_package.cpp:4]:
```suggestion
```

File: recipes/lodepng/all/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.0)
+project(cmake_wrapper)
+
+include(conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_library(lodepng "source_subfolder/lodepng.h" "source_subfolder/lodepng.cpp")
+
+include(GNUInstallDirs)
+install(TARGETS lodepng)
```
Comment (uilianries) [inline recipes/lodepng/all/CMakeLists.txt:10]:
```suggestion
install(TARGETS lodepng
        RUNTIME DESTINATION bin
        LIBRARY DESTINATION lib
        ARCHIVE DESTINATION lib)
```

explicit

**Comment by madebr** [comment]
@intelligide 

Check https://github.com/FrozenStormInteractive/conan-center-index/pull/2

File: recipes/lodepng/all/CMakeLists.txt (hunk 7 above)
Comment (intelligide) [reply recipes/lodepng/all/CMakeLists.txt:10]: GNUInstallDirs already defines standard GNU installation directories. (see https://cmake.org/cmake/help/latest/command/install.html#installing-targets)

Comment (uilianries) [reply recipes/lodepng/all/CMakeLists.txt:14]: I know, that's why I wrote "explicit".

Comment (madebr) [reply recipes/lodepng/all/CMakeLists.txt:14]: You include `GNUInstallDirs`, but don't use the variables.
The suggestion should use the `CMAKE_INSTALL_BINDIR` and `CMAKE_INSTALL_LIBDIR` variables.

File: recipes/lodepng/all/CMakeLists.txt [hunk 8]
```diff
@@ -11,5 +11,8 @@ endif()
 add_library(lodepng "source_subfolder/lodepng.h" "source_subfolder/lodepng.cpp")
 
 include(GNUInstallDirs)
-install(TARGETS lodepng)
+install(TARGETS lodepng
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBRARY}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_ARCHIVE})
```
Comment (madebr) [inline recipes/lodepng/all/CMakeLists.txt:17]:
```suggestion
    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
    LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}")
```

**Pull request merged**